### PR TITLE
Opentherm controller esp32

### PR DESCRIPTION
**Pull request by Cooler1989**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [ ] I have updated existing examples or added new ones (if applicable).
- [ ] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [ ] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [ ] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [ ] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [ ] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description
Please provide a clear and concise description of your changes, including the motivation behind these changes. The context is crucial for the reviewers.

#### Testing
Describe how you tested your changes.

### CODE REVIEW

File: examples/Cargo.toml [hunk 1]
```diff
@@ -57,6 +57,8 @@ static_cell         = { version = "2.1.0", features = ["nightly"] }
 usb-device          = "0.3.2"
 usbd-serial         = "0.2.2"
 
+opentherm_boiler_controller_lib = { version = "0.1.0", path = "/home/cooler1989/projects/rusty_boiler/boiler_controller_ot_lib", features = [] }
```
Comment (bugadani) [inline examples/Cargo.toml:60]: I'd like to ask you not to please do not add a dependency that lives on your filesystem.

File: examples/src/bin/embassy_rmt_opentherm.rs [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    }
+}
+
+const WIDTH: usize = 80;
+const RMT_CLK_DIV: u8 = 64;
+
+static RECEIVED_COUNT: Mutex<RefCell<Option<u32>>> =
+    Mutex::new(RefCell::new(None));
+
+static RECEIVED_DATA : critical_section::Mutex<RefCell<Option<[PulseCode; 48]>>> =
```
Comment (bugadani) [inline examples/src/bin/embassy_rmt_opentherm.rs:233]: If this isn't a different Mutex type than above, please do not specifiy the whole path to it.

**Review by bugadani — CHANGES_REQUESTED** [review]

File: examples/Cargo.toml (hunk 1 above)
Comment (Cooler1989) [reply examples/Cargo.toml:60]: I am sorry - it was not intended to be public. At least not yet. I was suppose to push I lt to private fork.

Comment (MabezDev) [reply examples/Cargo.toml:60]: This PR, or that dependency?

Comment (Cooler1989) [reply examples/Cargo.toml:60]: This whole PR. Can one delete it so it does not pollute this repository even in the history?

**Pull request closed without merging**